core/templates/dev/head/pages/exploration_editor/translation_tab/StateTranslationDirective.js

```diff
@@ -64,8 +68,45 @@ oppia.directive('stateTranslation', [
           $scope.stateDefaultOutcome = null;
           $scope.stateHints = [];
           $scope.stateSolution = null;
-          $scope.needsUpdateTooltipMessage = 'Audio needs update to match ' +
-            'text. Please record new audio.';
+
+          $scope.isVoiceoverModeActive = (
+            TranslationTabActiveModeService.isVoiceoverModeActive);
+          var isTranslatedTextRequired = function() {
+            return (TranslationTabActiveModeService.isVoiceoverModeActive() &&
+              TranslationLanguageService.getActiveLanguageCode() !== (
+                ExplorationLanguageCodeService.displayed));
+          };
+          $scope.getRequiredHtml = function(subtitledHtml) {
+            var html = null;
+            if (isTranslatedTextRequired()) {
+              var contentId = subtitledHtml.getContentId();
+              var activeLanguageCode = (
+                TranslationLanguageService.getActiveLanguageCode());
+              var writtenTranslations = (
+                ExplorationStatesService.getWrittenTranslationsMemento(
+                  $scope.stateName));
+              if (writtenTranslations.hasWrittenTranslation(
+                contentId, activeLanguageCode)) {
+                var writtenTranslation = (
+                  writtenTranslations.getWrittenTranslation(
+                    contentId, activeLanguageCode));
+                html = writtenTranslation.getHtml();
+              }
+            } else {
+              html = subtitledHtml.getHtml();
+            }
+            return html;
+          };
+
+          $scope.getEmptyContentMessage = function() {
+            if (TranslationTabActiveModeService.isVoiceoverModeActive()) {
+              return (
+                'There is no text available to voice-over, add a text ' +
```

Comment:
'There is no text available to voice-over or add a text translation.'

Maybe this is a better error message.

Reply:
I'm trying to ask users to add translations using the translation mode once they will add the text translation this tab will get populated with that data.

@nithusha21, Should I change this?

Reply:
If I remember correctly, I made this comment as you mention there is a translation mode, but you don't mention that there is a voice over mode. Also, I think if the voiceover mode is enabled, only the "voice-over" part of the message should be shown? (Why should we show that text translations error message in this situation?). 

Actually, after reading this error message again I am confused. Do you need to have a translation before you can voice over? That shouldn't be the case I think. If there is text, it can have voice-overs or translations irrespective of whether the other exists.

Reply:
> Do you need to have a translation before you can voice over?

If the selected language is the exploration language then the voice artist will see the main content otherwise they will see the translated content. 

We are assuming voice-artist is one who can do voice over for a text (they cannot add translations).  So if a voice artist is an expert in language X then he/she will see the translated text of exploration in X language and do a voice over for the same.

Reply:
Note: currently we don't have two roles but we have created these two modes to support these two roles.

Reply:
Oh I see, I think this is probably not the best. This removes the feature where the person could record audio for a lesson in a language of choice. Now he is forced to wait till a written translation is added. (What if he doesn't want to type one out himself, but is willing to record it?). I feel that one of the crucial features is being removed :(

What's your opinion?

Reply:
> This removes the feature where the person could record audio for a lesson in a language of choice. 
This still exists, but a user has to first translate the exploration in the selected language and then add a voiceover. I had a survey with all known oppia translators and all of us use to follow this pattern firstly write the translation in a piece of paper then record and upload audio. 
(We translated English to Hinglish which is way simple comparing translating English to Mandarin i.e, English can have a few text from English) 

About the future workflow of exploration translation:

The entire translation and voiceover work is going to be based on "Community contribution" work, so firstly a translator will do the entire translation (through suggestion) and then a voice artist will apply for a role for the same exploration and then they will add voiceover to the translated text.

Reply:
Hmmm, ok. I haven't done any translations myself, so I will agree with you (and the rest of the translators on this). Thanks for the explanation!